src/Uno.UI.Runtime.Skia.Wpf/Extensions/Storage/Pickers/FileOpenPickerExtension.cs

```diff
@@ -45,17 +45,34 @@ private async Task<StorageFile[]> OpenPickerAsync(bool multiple)
 			};
 
 			var filterBuilder = new StringBuilder();
-			filterBuilder.Append(string.Join("|", _picker.FileTypeFilter.Select(fileType => $"{fileType}|*{fileType}")));
+			var all = false;
+			var filterIndex = 0;
+			foreach (var fileType in _picker.FileTypeFilter)
+			{
+				if (fileType == "*")
+				{
+					all = true;
+				}
+				else
+				{
+					filterBuilder.Append($"|{fileType}|*{fileType}");
+					filterIndex++;
+				}
+			}
 
-			if (_picker.FileTypeFilter.Count > 1)
+			if (all)
+			{
+				filterBuilder.Append("|All|*");
```

Comment:
![image](https://user-images.githubusercontent.com/31348972/146938139-5a2eb3b5-d1cf-4333-a8e1-269ae5a636b7.png)

This is not necessarily the last item. But we need to keep track of its index to set `FilterIndex` properly.

Reply:
Thanks I hadn't considered this scenario